src/plugins/management/public/components/management_app/management_app.tsx

```diff
@@ -79,8 +82,19 @@ export const ManagementApp = ({ dependencies, history }: ManagementAppProps) =>
   );
 
   useEffect(() => {
+    // If workspace is enabled and user has entered workspace, hide advance settings and dataSource menu
```

Comment:
The change should happen inside workspace directory I suppose?

Reply:
Im okay, but if we put it in workspace plugin, we may need to declare to require `management` to get `management.sections.section.opensearchDashboards`, not sure if it's okay.

Reply:
That's OK, workspaces has dependency on saved objects management page as well.
Hide data source and advanced settings seems to be a workspace specific feature and better to change inside workspace plugin.

Reply:
Makes sense. Then I will add a required declaration.

Reply:
should it be optional dependency?

Reply:
I think we can use required. As this is an original required plugin, and mark it as required would not have any influences.

Reply:
As `management` is a plugin in OSD, it will always be installed?